src/main/scala/scalapb/json4s/JsonFormat.scala

```diff
@@ -27,9 +27,13 @@ case class JsonFormatException(msg: String, cause: Exception) extends Exception(
 case class Formatter[T](
   writer: (Printer, T) => JValue, parser: (Parser, JValue) => T)
 
+case class EnumFormatter[T](
+  writer: (Printer, T) => JValue, parser: (Parser, JValue) => Option[T])
+
+
 case class FormatRegistry(
   messageFormatters: Map[Class[_], Formatter[_]] = Map.empty,
-  enumFormatters: Map[EnumDescriptor, Formatter[EnumValueDescriptor]] = Map.empty,
+  enumFormatters: Map[EnumDescriptor, EnumFormatter[EnumValueDescriptor]] = Map.empty,
```

Comment:
Can you add a `@deprecated def this()` constructor that takes enumFormatters in the old signature and calls into this constructor? This way we will not break source compatibility.

Reply:
I don't think I can do this because of default args. Will the fact that `old` version of `FormatRegistry` will require `new` to be created is an issue as well? Also what about `registerEnumFormatter` ?

I'm not really familiar with `source compatibility`, so need your guidance here.